src/dns_resolver_sync.c

```diff
@@ -130,11 +145,20 @@ bool dns_resolver_is_lookup_complete(DNSRESOLVER_HANDLE dns_in)
                     case AF_INET:
                         /* Codes_SRS_dns_resolver_30_032: [ If dns_resolver_is_create_complete has returned true and the lookup process has succeeded, dns_resolver_get_ipv4 shall return the discovered IPv4 address. ]*/
                         dns->ip_v4 = EXTRACT_IPV4(ptr);
+#ifdef IPV6_ENABLED
```

Comment:
Why not enclosing just lines 151 to 155 instead?

Reply:
`dns->is_failed` is set to false below without IP_v4, I enclosed the correct lines now and added a second `dns->is_failed` check for IPv6 below